stable_baselines3/common/vec_env/vec_normalize.py

```diff
@@ -47,11 +50,12 @@ def __init__(
         self.norm_obs_keys = norm_obs_keys
         # Check observation spaces
         if self.norm_obs:
+            # Note: mypy doesn't take into account the sanity checks, which lead to several type: ignore...
             self._sanity_checks()
 
             if isinstance(self.observation_space, spaces.Dict):
                 self.obs_spaces = self.observation_space.spaces
-                self.obs_rms = {key: RunningMeanStd(shape=self.obs_spaces[key].shape) for key in self.norm_obs_keys}
+                self.obs_rms = {key: RunningMeanStd(shape=self.obs_spaces[key].shape) for key in self.norm_obs_keys}  # type: ignore[arg-type, union-attr]
```

Comment:
Why the type ignore? Shouldn't the new type annotations from the other PR now allow for a dict observation here?

Reply:
I think this one is due to mypy not taking the sanity checks into account.